nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+        List<String> kdfsForKeyedCipher = getKDFsForKeyedCipher();
+        if (kdf == null || !kdfsForKeyedCipher.contains(kdf.name())) {
+            validationResults.add(new ValidationResult.Builder().subject(KEY_DERIVATION_FUNCTION.getName())
+                    .explanation(KEY_DERIVATION_FUNCTION.getDisplayName() + " is required to be " + StringUtils.join(kdfsForKeyedCipher, ", ") + " when using algorithm " +
+                            encryptionMethod.getAlgorithm()).build());
+        }
+
+        return validationResults;
+    }
+
+    private List<String> getKDFsForKeyedCipher() {
+        List<String> kdfsForKeyedCipher = new ArrayList<>();
+        kdfsForKeyedCipher.add(KeyDerivationFunction.NONE.name());
+        for (KeyDerivationFunction k : KeyDerivationFunction.values()) {
+            if (k.isStrongKDF()) {
+                kdfsForKeyedCipher.add(k.name());
+            }
+        }
+        return kdfsForKeyedCipher;
+    }
+
+    private List<String> getKDFsForPBECipher(EncryptionMethod encryptionMethod) {
+        List<String> kdfsForPBECipher = new ArrayList<>();
+        for (KeyDerivationFunction k : KeyDerivationFunction.values()) {
+            // Add all weak (legacy) KDFs except NONE
+            if (!k.isStrongKDF() && !k.equals(KeyDerivationFunction.NONE)) {
+                kdfsForPBECipher.add(k.name());
+                // If this algorithm supports strong KDFs, add them as well
+            } else if ((encryptionMethod.isCompatibleWithStrongKDFs() && k.isStrongKDF())) {
+                kdfsForPBECipher.add(k.name());
+            }
+        }
+        return kdfsForPBECipher;
+    }
+
+    private Map<String, String> buildNewAttributes(FlowFile file, String atrList,
+                                                   EncryptContent.Encryptor encryptor,
+                                                   boolean isToBeEncrypted) throws Exception {
+
+        Map<String, String> oldAttributes = file.getAttributes();
+        Map<String, String> atrToWrite = new HashMap<>();
+        final String filenameAttr = CoreAttributes.FILENAME.key();
+        final String uuidAttr = CoreAttributes.UUID.key();
+
+        if (!StringUtils.isEmpty(atrList)) {
+            //TODO: check for spaces also
+            // Traverse comma separated list if provided
+            String[] attrs = atrList.split(",");
+            Set<String> atrSet = new HashSet<>(Arrays.asList(attrs));
+            for (String atr : atrSet) {
+                if (oldAttributes.containsKey(atr) && !atr.equals(filenameAttr) && !atr.equals(uuidAttr)) {
+                    String atrValue = oldAttributes.get(atr);
+                    String newAtrVal = (isToBeEncrypted) ? EncryptString.performEncryption(atrValue, encryptor) : EncryptString.performDecryption(atrValue, encryptor);
+                    atrToWrite.put(atr, newAtrVal);
+                }
+            }
+        } else {
+            //TODO: filename encryption check only for PGP algos.
```

Comment:
`filename` will be used in PGP algo so shouldn't be used for encryption or decryption only for PGP algos. I'm working on it.